crates/evm/fuzz/src/strategies/string.rs

```diff
@@ -0,0 +1,53 @@
+use crate::strategies::fixture_strategy;
+use alloy_dyn_abi::{DynSolType, DynSolValue};
+use proptest::{
+    arbitrary::any,
+    prelude::{prop, BoxedStrategy},
+    strategy::Strategy,
+};
+use rand::{distributions::Alphanumeric, thread_rng, Rng};
+
+/// The address strategy combines 2 different strategies:
+/// 1. A random addresses strategy if no fixture defined for current parameter.
+/// 2. A fixture based strategy if configured values for current parameters.
+/// If fixture is not a valid type then an error is raised and test suite will continue to execute
+// with random strings.
+///
+///
+/// For example:
+/// To define fixture for `person` fuzzed parameter, return an array of possible values from
+/// `function fixture_person() public returns (string[] memory)`.
+/// Use `person` named parameter in fuzzed test in order to create a custom strategy
+/// `function testFuzz_personValue(string memory person)`.
+#[derive(Debug, Default)]
+pub struct StringStrategy {}
+
+impl StringStrategy {
+    /// Create a new string strategy.
+    pub fn init(fixtures: Option<&[DynSolValue]>) -> BoxedStrategy<DynSolValue> {
+        let value_from_fixture = |fixture: Option<&DynSolValue>| {
+            if let Some(fixture) = fixture {
+                if let Some(fixture) = fixture.as_str() {
+                    return DynSolValue::String(fixture.to_string());
+                }
+            }
```

Comment:
```suggestion
            if let Some(val @ DynSolValue::String(_)) = fixture {
                return val.clone()
            }
```

I believe the same is possible for address and bytes strats. Wondering why do we have different logic for value-generation when no fixtures are added vs when fixture returns incorrect value?

Reply:
thing is that we don't know if fixture is of valid type until the point we generate value from strategy, so cannot use default `DynSolValue::type_strategy` in this case. Wonder if it would be better to not generate random values and just let fuzzer panic if wrong type (so user will be aware right away that configured fixtures are not used in campaign) instead printing errors, wdyt?

Reply:
I think we should change flow a bit so that `ContractRunner::fuzz_fixtures` flattens all arrays right away and keeps sets of `DynSolValue` for each func name, enforcing that all of them have same type.

That way, when creating strategy we can validate that all matching fixtures for given parameter have the same value as the parameter itself. That way we can just panic if we got a fixture of invalid type and this should be safe

Reply:
yeah, agree, I just changed code to let fuzzer panic if the param type is not of fixtures type. The risk here was when fixtures were matched by param name only (as in someone could have been declaring a `string[]` fixture for fuzzing an `address` type) but now since we explicitly require functions prefixed with `fixture` I don't think that's a problem anymore - this also simplified code a lot.